src/main/java/javax/validation/valueextraction/ExtractedValue.java

```diff
@@ -27,4 +28,20 @@
 @Documented
 public @interface ExtractedValue {
 
+	/**
+	 * Returns the type of the value extracted by the {@link ValueExtractor}. If not set, returns
+	 * {@code Void.class}, meaning the type will be automatically inferred from the type argument
+	 * of the parameterized type.
+	 * <p>
+	 * {@code type} is used to define {@code ValueExtractor}s for non parameterized type e.g.
+	 * {@link OptionalInt}.
+	 * <p>
+	 * {@code type} may not be used when {@code ExtractedValue} is defined on the type parameter of
+	 * a parameterized container type. A {@code ValueExtractorDefinitionException} will be thrown in
+	 * this case.
+	 *
+	 * @return the type of the value extracted by the {@link ValueExtractor}
```

Comment:
I'd not make this a link, you did it already in the method docs. The guideline says to do it only once. Here I'd just put "value extractor"